File: Wacomi.API/Models/CircleTopicComment.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Wacomi.API.Models
{
    public class CircleTopicComment : IAppUserLinkable, IDataItemWithSinglePhoto
    {
        public CircleTopic CircleTopic { get; set;}
        [Required]
        public int CircleTopicId{ get; set;}
        public int CircleId{ get; set;}
        [Required]
        [MaxLength(2000)]
        public string Comment{ get; set;}
        public Photo Photo { get;set;}
        public int? PhotoId {get; set;}
        public virtual ICollection<CircleTopicCommentReply> Replies{ get; set;}
        public AppUser AppUser {get;set;}
        public int? AppUserId {get;set;}
        public bool IsActive {get; set;} = true;
        public int Id {get; set;}
        public DateTime DateCreated {get; set;} = DateTime.Now;
        public DateTime DateUpdated {get; set;} = DateTime.Now;
    }
}

File: Wacomi.API/ClientApp/src/app/_models/CircleTopicComment.ts
import { Photo } from "./Photo";
import { IDataItem } from "./IDataItem";
import { IAppUserLinkable } from "./IAppUserLinkable";
import { ShortComment } from "./ShortComment";

// public int Id { get; set;}
//         public int CircleTopicId{ get; set;}
//         public string Comment{ get; set;}
//         public PhotoForReturnDto Photo { get;set;}
//         public int? PhotoId {get; set;}
//         public int ReplyCount{ get; set;}
//         public AppUserForReturnDto AppUser {get;set;}
//         public int? AppUserId {get;set;}
//         public DateTime DateCreated {get; set;}
//         public DateTime DateUpdated {get; set;}
        
export interface CircleTopicComment extends IAppUserLinkable, IDataItem {
    circleTopicId: number;
    comment:string;
    photo: Photo;
    photoId: number;
    replyCount:number;

    //Client side only
    displayReplies?: boolean;
    //topicReplies: TopicReply[];
    topicReplies?: ShortComment[];
}
